src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9995,7 +9998,12 @@ void Unit::ProcSkillsAndReactives(bool isVictim, Unit* procTarget, uint32 typeMa
     if (typeMask & MELEE_BASED_TRIGGER_MASK && procTarget)
     {
         // Update skills here for players
-        if (GetTypeId() == TYPEID_PLAYER)
+        // only when you are not fighting other players or their pets/totems (pvp)
+        if (GetTypeId() == TYPEID_PLAYER &&
+                procTarget->GetTypeId() != TYPEID_PLAYER &&
+                !(procTarget->IsTotem() && procTarget->ToTotem()->GetOwner()) &&
```

Comment:
GetOwner checks are pointless here, a totem or pet cannot exist without owher

Reply:
From the source files, GetOwner `Player* GetOwner() const;` Does it mean that any NPC can be passed as a Player pointer? The idea's to check that pet/totem belongs to a player and not to NPC.

Reply:
Totem inherits Minion https://github.com/TrinityCore/TrinityCore/blob/3ac1992afbeb1cd7669b300b8ee4131d39f35595/src/server/game/Entities/Creature/TemporarySummon.h#L76 and thats Unit*, not Player*

`Pet` cannot be created for a creature owner at all, so if you have one you know its owned by a player
* pretty sure totems dont grant weapon skills in any case, not just when owned by a player

Reply:
Yeah, you are right about Totems, I gotta check if there's another way to find a player who summoned it.

But can't agree with you about pets:
https://github.com/TrinityCore/TrinityCore/blob/3ac1992afbeb1cd7669b300b8ee4131d39f35595/src/server/game/Entities/Unit/Unit.h#L1656
https://github.com/TrinityCore/TrinityCore/blob/3ac1992afbeb1cd7669b300b8ee4131d39f35595/src/server/game/Entities/Pet/Pet.h#L148

Reply:
The point I was trying to make here is that no totem should grant weapon skillups, not just player owned one

Reply:
@Shauren I couldn't find any confirmation on the internet about how it really should work with weapon skill leveling proc chance while fighting against NPC totems. Though I feel like in ZF on Classic retail I was able to level up weapon skills by hitting totems, but currently, I don't have a subscription to check that.
If you still want to disable proc chance for everyone except NPC mobs then I could simplify the weapon proc if-check. Otherwise please check my comment below where I described how the code works.